boards/apollo3/lora_things_plus/src/tests/atecc508a.rs

```diff
@@ -0,0 +1,86 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2023.
+
+//! Test that the Atecc508a works
+
+use crate::tests::run_kernel_op;
+use crate::ATECC508A;
+use kernel::debug;
+
+#[test_case]
+fn read_config() {
+    run_kernel_op(100_000);
+
+    debug!("check run ATECC508A config... ");
+    run_kernel_op(100);
+
+    unsafe {
+        let atecc508a = ATECC508A.unwrap();
+
+        atecc508a.read_config_zone().unwrap();
+    }
+
+    run_kernel_op(150_000);
+    debug!("    [ok]");
+    run_kernel_op(100);
+}
+
+#[test_case]
+fn setup_and_lock_tock_config() {
+    run_kernel_op(100_000);
+
+    debug!("Lock the Tock config...");
+    run_kernel_op(100);
+
+    unsafe {
+        let atecc508a = ATECC508A.unwrap();
+
+        atecc508a.read_config_zone().unwrap();
+        run_kernel_op(150_000);
+
+        if atecc508a.device_locked() {
+            debug!("    [ok] - Already locked");
+            return;
+        }
+
+        debug!("This can not be undone!");
```

Comment:
I don't like that we're permanently locking a config without user input. Imagine a user doesn't have their serial port open -- they may easily just leave it connected after flashing. Is locking required to use the device, and if so, how is this test/function even run? Adding some docs here, and perhaps waiting until a button is pushed would be good.

Reply:
Locking is required to use the device at all. The test is run as part of the usual `make test` process.

I can add some more docs to explain that as well. The board does have a button, but mine doesn't work (no idea why) so I can't easily test waiting on a button

Reply:
Bummer. I guess waiting for a character on the UART is too hard in the kernel?

I mis-burned some expensive clock chips before by accident, and it's incredibly frustrating when fuses / configs are burnt without explicit, all-caps, triple-checked user confirmation. In the best case it's "whatever", worse it means re-ordering and waiting for a new chip, and for some people the $4.95 + shipping might also be a not-insignificant amount of money.

So I _really_ would like to see us have some form of user input beforehand. :confused:

Reply:
That's fair... But I don't really have a easy way to ask for input before running the kernel tests.

My feeling at the moment is that a user must explicitly enable this feature and plug the device into the board. At which point hopefully they know we have to lock the device otherwise nothing on it works